scripts/ekf_logs/logs_evaluations/evaluations/gps_ekf_comparison.py

```diff
@@ -0,0 +1,295 @@
+        sensor_timestamp = np.array([log.timestamp for log in context.imu_logs])
+        state_timestamp = np.array([log.timestamp for log in context.state_logs])
+
+        nearest_state_indices = nearser_values(state_timestamp, sensor_timestamp)
+        rotations_inv = np.array([context.state_logs[i].data.attitude.inv() for i in nearest_state_indices])
+
+        north = np.empty(len(context.imu_logs))
+        east = np.empty(len(context.imu_logs))
+
+        for i in range(len(north)):
+            vector = rotations_inv[i].apply(context.imu_logs[i].data.accel.as_array())
+            north[i] = vector[0]
+            east[i] = vector[1]
+
+        return north, east
+
+    def _iir_filter(self, accels):
+        result = np.empty(len(accels))
+
+        result[0] = accels[0]
+
+        for i in range(1, len(result)):
+            result[i] = 0.1 * accels[i] + 0.9 * result[i-1]
+
+        return result
+
+    def _draw_gps_pos(self, ax, context: StudyContext):
+        x = np.array([self.calculate_gps_x(log) for log in context.gps_logs])
+        y = np.array([self.calculate_gps_y(log) for log in context.gps_logs])
+
+        ax.scatter(x, y, c="#f542e9", marker='s', label="GPS position")
+
+        for i in range(len(context.gps_logs)):
+            ax.annotate(
+                self.format_timestamp(context.gps_logs[i].timestamp),
+                (x[i], y[i]),
+                ha="center",
+                xytext=(0, -12),
+                textcoords='offset points'
+            )
+
+    def _draw_ekf_state_pos(self, ax, context: StudyContext):
+        x = np.array([log.data.position.east for log in context.state_logs])
+        y = np.array([log.data.position.north for log in context.state_logs])
+
+        ax.scatter(x, y, s=3, label="EKF state position")
+
+    def _draw_gps_extrapolation(self, ax, context: StudyContext):
+        color = "#1ccc18"  # Green
+
+        extrapolated_x = np.empty(len(context.gps_logs) - 2)
+        extrapolated_y = np.empty(len(context.gps_logs) - 2)
+
+        # Calculate extrapolation
+        for i in range(1, len(context.gps_logs) - 1):
+            prev_log = context.gps_logs[i-1]
+            curr_log = context.gps_logs[i]
+            next_log = context.gps_logs[i+1]
+
+            if (curr_log.timestamp == prev_log.timestamp):
+                # Truncating the last element
+                extrapolated_x = extrapolated_x[:len(extrapolated_x) - 2]
+                extrapolated_y = extrapolated_y[:len(extrapolated_y) - 2]
+                continue
+
+            dt = curr_log.timestamp - prev_log.timestamp
+            extrapolation_dt = next_log.timestamp - curr_log.timestamp
+
+            prev_pos_x = self.calculate_gps_x(context.gps_logs[i-1])
+            prev_pos_y = self.calculate_gps_y(context.gps_logs[i-1])
+
+            curr_pos_x = self.calculate_gps_x(context.gps_logs[i])
+            curr_pos_y = self.calculate_gps_y(context.gps_logs[i])
```

Comment:
If there is no `prev_vel_x` then it may be just:
```suggestion
            prev_x = self.calculate_gps_x(context.gps_logs[i-1])
            prev_y = self.calculate_gps_y(context.gps_logs[i-1])

            curr_x = self.calculate_gps_x(context.gps_logs[i])
            curr_y = self.calculate_gps_y(context.gps_logs[i])
```